### DISCUSSION TITLE
Input: The 'x' to clear renders incorrectly

### DISCUSSION THREAD

**Discussion by martinkrulltott** [dhis2/notes#234] [Q&A] [answered]
The 'x' that's used to clear an `Input` renders incorrectly. On mouse leave the very bottom of the x stays on the screen while the rest of the x is hidden. Vice versa can also be seen, where the x is shown but the very bottom is missing. (Tested in Chrome OSX by @martinkrulltott and Chrome Linux / Pop OS by @janhenrikoverland ).

_in action_
https://user-images.githubusercontent.com/12590483/108480976-e4d93380-7297-11eb-8cd4-057949bcdd8d.mp4

_very bottom is missing_
![image](https://user-images.githubusercontent.com/12590483/108481230-3aaddb80-7298-11eb-8a31-7a2f519b877f.png)

_regular / expected style_
![image](https://user-images.githubusercontent.com/12590483/108481251-40a3bc80-7298-11eb-98f7-8470d7ac589f.png)

**Comment by Mohammer5** [comment]
This isn't really a Q/A issue, right?
It's a bug report, you could just create an issue in Jira.

**Reply by martinkrulltott** [reply]
IDK, maybe its per design? Or out of scope? Or something that needs to be discussed first? But if you want me to create a ticket for it then I will.

**Reply by martinkrulltott** [reply]
https://jira.dhis2.org/browse/LIBS-128

**Reply by Birkbjo** [reply]
I'm unsure if this is a UI-bug? It's a browser feature when `type='search'`, so it's not something we support.

**Reply by martinkrulltott** [reply]
I thought we were supposed to support / use it though, @cooper-joe ?

**Reply by Birkbjo** [reply]
I mean, it's not something that is implemented by us, it doesn't even show up in Firefox. If it's something we want to support, maybe it's something we should implement and control ourselves, so it's consistent across browsers?

**Reply by cooper-joe** [reply]
> I thought we were supposed to support / use it though, @cooper-joe ?

Yes, using `type="search"` on a `ui` `input` is supported and should be used. Different browsers handle this type declaration differently (for example, Firefox not showing a clear control). 

The approach right now, aligned with our approach on accessibility, is to not block native browser functionality. This counts as native browser functionality. So the answer to the question, "does `ui` support this?" is yes.

This issue doesn't block the native browser functionality, so it's not a functional bug. I'd categorise it as a visual bug where `ui` is not correctly displaying native browser controls.

> maybe it's something we should implement and control ourselves, so it's consistent across browsers?

I think we may want to do this eventually, but this is another discussion.

**Reply by varl** [reply]
Since I cannot mark a nested comment as the answer I'll quote @cooper-joe in a new comment and mark that as the answer.

**Comment by varl** [comment]
> I thought we were supposed to support / use it though, @cooper-joe ?
> 
> Yes, using `type="search"` on a `ui` `input` is supported and should be used. Different browsers handle this type declaration differently (for example, Firefox not showing a clear control).
> 
> The approach right now, aligned with our approach on accessibility, is to not block native browser functionality. This counts as native browser functionality. So the answer to the question, "does `ui` support this?" is yes.
> 
> This issue doesn't block the native browser functionality, so it's not a functional bug. I'd categorise it as a visual bug where `ui` is not correctly displaying native browser controls.
> 
> > maybe it's something we should implement and control ourselves, so it's consistent across browsers?
> 
> I think we may want to do this eventually, but this is another discussion.

What @cooper-joe wrote here is correct. We recommend using the native browser functionality here, even though it is not rendered perfectly in every browser on every platform. That is simply not in the current scope because of our resource constraints.

**Reply by martinkrulltott** [reply]
> I'd categorise it as a visual bug where ui is not correctly displaying native browser controls.

If it's a bug it should be raised in JIRA, even though it might be out of scope right now?
I'm not arguing that this is an important thing to fix, but simply ignoring it doesn't seem like the proper solution or "Answer" to this Q&A ticket.

**Reply by Mohammer5** [reply]
> I'd categorise it as a visual bug where ui is not correctly displaying native browser controls.

I wouldn't necessarily say it's a bug of our UI library. It's the browsers that cause the pixel glitch. And - at least right now - we're not planning on overwriting native browser controls.

> If it's a bug it should be raised in JIRA, even though it might be out of scope right now?

We could still create a JIRA issue for this, I'm not sure what the right category would be though, as it's not really a bug on our side, we haven't decided on implementing a feature that would replace this and there's no task to be done (cc @varl). To me, the appropriate category would be "discussion", which is the purpose of this GH discussions board.

I'd probably just consider this "done" / "answered": It's the browsers fault, we use the native browser behavior, so the only thing we could do is to file a bug report int he chrome/chromium report system

**Reply by martinkrulltott** [reply]
> I wouldn't necessarily say it's a bug of our UI library. It's the browsers that cause the pixel glitch.

Attached a video below of a native HTML `input` vs a ui `InputField`, both with `type="search"`.

Wherever the bug lies, it's not affecting the native `input`. Regardless though, IMO, a bug is a bug and should be treated as such.

The video is based on the [ItemSelector in Analytics](https://github.com/dhis2/analytics/blob/bcf400f6f43d0794213070d52b0bd6db7aa61ea6/src/components/DataDimension/ItemSelector.js#L57), with the addition of a native `input`.

```
<input type="search" />
<InputField
   value={searchTerm}
   onChange={({ value }) => setSearchTerm(value)}
   placeholder={i18n.t('Search by data item name')}
   dataTest={`${dataTest}-filter-input-field`}
   dense
   initialFocus
   type={'search'}
/>
```

https://user-images.githubusercontent.com/12590483/109469961-0d6ae580-7a6f-11eb-8528-5cf3c0badddb.mov

**Reply by varl** [reply]
Yes, the correct course of action would be to report it in JIRA.

The JIRA issue may be closed with "won't fix" as resolution.

The issue might be fixed regardless when peripheral work is done on native form controls.

If a patch that fixes it is submitted in a PR it would get merged. Contributions are welcome.

**Reply by Mohammer5** [reply]
Ah I didn't understand that it's a bug that's somehow caused by us :ok_hand:

**Comment by varl** [comment, marked as answer]
https://jira.dhis2.org/browse/LIBS-128
